### ISSUE TITLE
Improve ETH Spend to sync return more quickly or to become entirely async

### ISSUE THREAD

**Issue by placer14** [OpenBazaar/openbazaar-go#1783]
As a stopgap, we have allowed the ETH Spend method to take as long as 3 minutes waiting for the transaction to be mined by the network as there is no other way to ensure whether the transaction is successful or not until then. We would either improve this to emit a websocket message result for ETH transactions which would allow the Spend request to return the client response more quickly while still tracking the result of the Spend.

This improvement should be discussed with desktop and mobile client integration teams.

**Comment by rmisio** [comment]
> We would either improve this to emit a websocket message result for ETH transactions which would allow the Spend request to return the client response more quickly while still tracking the result of the Spend.

If we do that, I would make that configurable via an api parameter. Some clients may prefer to wait than have a disjointed response.

(see `async` on `ob/fetchprofiles`)

Secondly, for the socket option, the client should be allowed to pass in a socket id that the socket will return and that would be used as the mapping between the API call and the socket response. We found this pattern worked well as opposed to the server generating and sending the socket id in the API response, since we found that there were sometimes race conditions where the socket arrived before the API returned.

(see `asyncID` on `ob/fetchprofiles`)
